lib/widgets/sign_in_widget.dart

```diff
@@ -27,16 +74,38 @@ class _SignInState extends State<SignIn> {
             child: Column(
               mainAxisSize: MainAxisSize.min,
               children: [
-                TextField(
-                  style: Theme.of(context).textTheme.bodyText1,
-                  decoration: InputDecoration(
-                    hintText: AppText.emailText,
-                  ),
-                ),
-                TextField(
-                  style: Theme.of(context).textTheme.bodyText1,
-                  decoration: InputDecoration(
-                    hintText: AppText.passwordText,
+                Form(
+                  key: formKey,
+                  child: Column(
+                    children: [
+                      TextFormField(
+                        validator: (value) {
```

Comment:
you can consider using this package for form validation
https://pub.dev/packages/form_validator